src/Database/index.js

```diff
@@ -175,14 +177,37 @@ export default class Database {
     this._ensureInAction(
       `Database.unsafeResetDatabase() can only be called from inside of an Action. See docs for more details.`,
     )
-    // Doing this in very specific order:
-    // First kill actions, to ensure no more traffic to adapter happens
-    // then clear the database
-    // and only then clear caches, since might have had queued fetches from DB still bringing in items to cache
-    this._actionQueue._abortPendingActions()
-    await this.adapter.unsafeResetDatabase()
-    this._unsafeClearCaches()
-    this._resetCount += 1
+    try {
+      this._isBeingReset = true
+      // First kill actions, to ensure no more traffic to adapter happens
+      this._actionQueue._abortPendingActions()
+
+      // Kill ability to call adapter methods during reset (to catch bugs if someone does this)
+      const { adapter } = this
+      const ErrorAdapter = require('../adapters/error').default
+      this.adapter = (new ErrorAdapter(): any)
+
+      // Check for illegal subscribers
+      if (this._subscribers.length) {
+        // eslint-disable-next-line no-console
+        console.error(
+          'Application error! 1 Database subscriber was detected during database.unsafeResetDatabase() call. App should not hold onto subscriptions or Watermelon objects while resetting database.',
```

Comment:
Will it always be `1`? Or should it be `{this._subscribers.length}`?

Reply:
Oof. Brain fart — thanks for catching this.

Reply:
I think you need to change it in 1 of the tests as well.

Reply:
What do you mean? https://github.com/Nozbe/WatermelonDB/pull/671/files#diff-1c7bf3678f7379f801755def5d7c2facR109 is correct - checks if there'e exactly 1 subscriber too many

Reply:
Right, that's true. CI is failing and that was my first guess.